test/crossbrowser/src/browser/api.ts

```diff
@@ -0,0 +1,362 @@
+
+  room.participants.forEach((participant: any) => {
+    add(participant);
+    participant.tracks.forEach(add);
+  });
+  sendRoomEvents(room);
+
+  return {
+    result: serializeRoom(room)
+  };
+}
+
+/**
+ * Create a {@link LocalTrack}.
+ * @param {Array<*>} args
+ * @returns {Promise<object>}
+ */
+export async function createLocalTrack(args: any): Promise<any> {
+  const [ kind, options ] = args;
+  const createLocalTrack: (options: any) => Promise<any> = {
+    audio: (options: any) => Twilio.Video.createLocalAudioTrack(options),
+    data: (options: any) => new Twilio.Video.LocalDataTrack(options),
+    video: (options: any) => Twilio.Video.createLocalVideoTrack(options)
+  }[kind];
+  let localTrack: any;
+
+  try {
+    localTrack = await createLocalTrack(options);
+  } catch (e) {
+    return {
+      error: {
+        message: e.message
+      }
+    };
+  }
+  add(localTrack);
+
+  return {
+    result: serializeLocalTrack(localTrack)
+  };
+}
+
+/**
+ * Create an array of {@link LocalTrack}s.
+ * @param {Array<*>} args
+ * @returns {Promise<object>}
+ */
+export async function createLocalTracks(args: any): Promise<any> {
+  const [ options ] = args;
+  let localTracks: Array<any>;
+
+  try {
+    localTracks = await Twilio.Video.createLocalTracks(options);
+  } catch (e) {
+    return {
+      error: {
+        message: e.message
+      }
+    };
+  }
+  localTracks.forEach(add);
+
+  return {
+    result: localTracks.map(serializeLocalTrack)
+  };
+}
+
+/**
+ * Disconnect from a {@link Room}.
+ * @param {ResourceID} target - Resource ID of the {@link Room}
+ * @returns {object}
+ */
+export function disconnect(target: ResourceID): any {
+  const room: any = lookup(target);
+  if (!room) {
+    return {
+      error: {
+        message: 'Room not found'
+      }
+    };
+  }
+  room.disconnect();
+  const serializedRoom = serializeRoom(room);
+  remove(room);
+
+  return {
+    result: serializedRoom
+  };
+}
+
+/**
+ * Get {@link Room} stats.
+ * @param {ResourceID} target - Resource ID of the {@link Room}.
+ * @returns {Promise<object>}
+ */
+export async function getStats(target: ResourceID): Promise<any> {
+  const room: any = lookup(target);
+  if (!room) {
+    return {
+      error: {
+        message: 'Room not found'
+      }
+    };
+  }
+
+  try {
+    return {
+      result: await room.getStats()
+    };
+  } catch (e) {
+    return {
+      error : {
+        message: e.message
+      }
+    };
+  }
+}
+
+/**
+ * Publish a {@link LocalTrack} to a {@link Room}.
+ * @param {ResourceID} target - {@link LocalParticipant} resource ID
+ * @param {Array<*>} args
+ * @returns {Promise<object>}
+ */
+export async function publishTrack(target: ResourceID, args: any): Promise<any> {
+  const localParticipant: any = lookup(target);
+  if (!localParticipant) {
+    return {
+      error: {
+        message: 'Participant not found'
+      }
+    };
+  }
+
+  const localTrack: any = lookup(args[0]);
```

Comment:
We should type `args`, I think. Otherwise how do we know this contract is fulfilled? (i.e., `args[0]` is what we need it to be?).